linalg/solvers.hpp

```diff
@@ -895,13 +895,14 @@ class DirectSubBlockSolver : public Solver
    mutable Array<int> local_dofs;
    mutable Vector sub_rhs;
    mutable Vector sub_sol;
-   Array<DenseMatrixInverse> block_solvers;
+   DenseMatrixInverse *block_solvers;
```

Comment:
Would it work to use `std::vector<DenseMatrixInverse>` instead of a pointer?

Reply:
Personally I tend to prefer `std::vector` over `new[]` because then there is no need for a non-trivial destructor, no risk of leaking memory, it's automatically copyable and movable, etc.